### PR TITLE
Create generate-random-pkmn.c

### PR DESCRIPTION
**Pull request by FM1337**

### CODE REVIEW

File: src/universal/generate-random-pkmn.c [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    int maxSpecies = 0;
+    int maxMoves = 0;
+    int maxBalls = 0;
+    int maxItems = 0;
+    int maxAbility = 0;
+
+    gui_warn("Please enter how many pokemon", "you would like to generate!");
+
+    int randomizedCount = 0;
+    gui_numpad(&randomizedCount, "You are entering how many randomized\npokemon you would like to generate", 3);
```
Comment (piepie62) [inline src/universal/generate-random-pkmn.c]: This is a much longer "What?" message than normal. Second-person is also kind of odd here, in my opinion

File: src/universal/generate-random-pkmn.c [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        case 10:
+        case 11:
+        case 12:
+        case 7:
+        case 8:
+            gen = GEN_FOUR;
+            maxSpecies = 493;
+            maxBalls = 18;
+            maxMoves = 467;
+            if(version == 11 || version == 10){
```
Comment (piepie62) [inline src/universal/generate-random-pkmn.c]: Why not just do these in the cases themselves?

File: src/universal/generate-random-pkmn.c [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            break;
+        case 42:
+        case 43:
+            gen = GEN_LGPE;
+            maxSpecies = 153;
+            break;
+    }
+
+    if (randomizedCount > maxSpecies) {
+        char part2[256];
```
Comment (piepie62) [inline src/universal/generate-random-pkmn.c]: Explicitly zero-initialize this

File: src/universal/generate-random-pkmn.c [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                if (pokemonID == 152) {
+                    pokemonID = 808;
+                } else if (pokemonID == 153) {
+                    pokemonID = 809;
+                }
+            }
+            pkx_generate(data, pokemonID);
+            // Shiny or not
+            pkx_set_value(data, gen, SHINY, shiny);
+             pkx_set_value(data, gen, LEVEL, randLevel);
```
Comment (piepie62) [inline src/universal/generate-random-pkmn.c]: Extra space here

File: src/universal/generate-random-pkmn.c [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                pkx_set_value(data, gen, MOVE, 2, (rand() % maxMoves)+1);
+                pkx_set_value(data, gen, MOVE, 3, (rand() % maxMoves)+1);
+                pkx_set_value(data, gen, ITEM, randItem);
+                pkx_set_value(data, gen, ABILITY, randAbility);
+            }
+            pkx_set_value(data, gen, POKERUS, (rand() %  255)+1, (rand() %  255)+1) ;
+            sav_inject_pkx(data, gen, i / 30, i % 30, 0);
+        }
+        free(data);
+    sav_box_encrypt();
```
Comment (piepie62) [inline src/universal/generate-random-pkmn.c]: Weird indentation

File: src/universal/generate-random-pkmn.c (hunk 2 above)
Comment (FM1337) [reply src/universal/generate-random-pkmn.c]: I want to keep all the gens grouped together.

File: src/universal/generate-random-pkmn.c (hunk 1 above)
Comment (FM1337) [reply src/universal/generate-random-pkmn.c]: fixed

File: src/universal/generate-random-pkmn.c (hunk 3 above)
Comment (FM1337) [reply src/universal/generate-random-pkmn.c]: fixed

File: src/universal/generate-random-pkmn.c (hunk 4 above)
Comment (FM1337) [reply src/universal/generate-random-pkmn.c]: fixed

File: src/universal/generate-random-pkmn.c (hunk 5 above)
Comment (FM1337) [reply src/universal/generate-random-pkmn.c]: fixed

File: src/universal/generate-random-pkmn.c (hunk 2 above)
Comment (piepie62) [reply src/universal/generate-random-pkmn.c]: Fair enough

**Comment by FM1337** [comment]
should be good now?

File: src/universal/generate-random-pkmn.c (hunk 3 above)
Comment (piepie62) [reply src/universal/generate-random-pkmn.c]: Was done incorrectly

Comment (FM1337) [reply src/universal/generate-random-pkmn.c]: fixed in the current one

**Comment by piepie62** [comment]
Looks good to me

**Comment by FM1337** [comment]
Currently broken as of PKSM commit [d04bba0](https://github.com/FlagBrew/PKSM/commit/d04bba04d31b9afc502fc2c7501494f5fbf688ac)

**Comment by piepie62** [comment]
Should be fixed

**Comment by piepie62** [comment]
Looks good to me, thanks

**Pull request merged**